src/v/cluster/partition_balancer_planner.cc

```diff
@@ -184,170 +135,192 @@ std::optional<size_t> partition_balancer_planner::get_partition_size(
+        distinct_from(rrs.unavailable_nodes_vec)));
 
-    auto constraints = partition_constraints(
+    return partition_constraints(
       assignments.id,
       topic_metadata.get_configuration().replication_factor,
-      allocation_constraints);
+      std::move(allocation_constraints));
+}
 
-    size_t full_node_amount = get_full_nodes_amount(assignments.replicas, rrs);
-    for (size_t i = 0; i <= full_node_amount; ++i) {
-        auto stable_replicas = get_stable_replicas(
-          assignments.replicas, rrs, i);
+result<allocation_units> partition_balancer_planner::get_reallocation(
+  const model::ntp& ntp,
+  const partition_assignment& assignments,
+  size_t partition_size,
+  partition_constraints constraints,
+  const std::vector<model::broker_shard>& stable_replicas,
+  reallocation_request_state& rrs) {
+    vlog(
+      clusterlog.debug,
+      "trying to find reallocation for ntp {} with stable_replicas: {}",
+      ntp,
+      stable_replicas);
 
-        if (stable_replicas.size() == assignments.replicas.size()) {
-            break;
-        }
+    auto stable_assigments = partition_assignment(
+      assignments.group, assignments.id, stable_replicas);
 
-        auto stable_assigments = partition_assignment(
-          assignments.group, assignments.id, stable_replicas);
-
-        auto reallocation = _partition_allocator.reallocate_partition(
-          constraints, stable_assigments);
-
-        auto ntp = model::ntp(
-          topic_metadata.get_configuration().tp_ns.ns,
-          topic_metadata.get_configuration().tp_ns.tp,
-          assignments.id);
-        if (reallocation.has_value()) {
-            rrs.moving_partitions.insert(ntp);
-            rrs.planned_movement_disk_size += partition_size;
-            for (const auto r :
-                 reallocation.value().get_assignments().front().replicas) {
-                if (
-                  std::find(stable_replicas.begin(), stable_replicas.end(), r)
-                  == stable_replicas.end()) {
-                    rrs.assigned_reallocation_sizes[r.node_id]
-                      += partition_size;
-                }
-                if (
-                  std::find(
-                    assignments.replicas.begin(), assignments.replicas.end(), r)
-                  == assignments.replicas.end()) {
-                    rrs.node_released_disk_size[r.node_id] += partition_size;
-                }
+    auto reallocation = _partition_allocator.reallocate_partition(
+      std::move(constraints), stable_assigments);
+
+    if (!reallocation) {
+        vlog(
+          clusterlog.debug,
+          "attempt to find reallocation for ntp {} with "
+          "stable_replicas: {} failed, error: {}",
+          ntp,
+          stable_replicas,
+          reallocation.error().message());
+
+        return reallocation;
+    }
+
+    rrs.moving_partitions.insert(ntp);
+    rrs.planned_movement_disk_size += partition_size;
```

Comment:
i know this is out of this PR scope but maybe we should rename this variable to simply `planned_moves_bytes` or `planned_moves_size` ?

Reply:
sure, no problem